### DISCUSSION TITLE
question about studying and compiling this project from source

### DISCUSSION THREAD

**Discussion by Jarrodsz** [mrkai77/Loop#26] [Q&A] [answered]
I have been looking for a great application to implement some features i was looking for and yours is awesome.
I hope i may ask some advice dev to dev on how i could run this code in xcode since i get a lot of errors. 

Im on 13.5.1 (22G90) 

For example xcode complains about 
```
var nextPreviewDirection: WindowDirection {
        switch self {
        case .noAction:             .topHalf
        case .topHalf:              .topRightQuarter
        case .topRightQuarter:      .rightHalf
        case .rightHalf:            .bottomRightQuarter
        case .bottomRightQuarter:   .bottomHalf
        case .bottomHalf:           .bottomLeftQuarter
        case .bottomLeftQuarter:    .leftHalf
        case .leftHalf:             .topLeftQuarter
        case .topLeftQuarter:       .maximize
        case .maximize:             .noAction
        default:                    .noAction
        }
    }``` 

This section.

Do i need to install anything special or setup my xcode in a way so i could just run test and expiriment with the code to learn how to do certain things? 

Kind regards

**Comment by mrkai77** [comment, marked as answer]
You will need Xcode 15 to compile Loop. Also, once you finish implementing the features you want, please feel welcome to pull request your changes :)

**Reply by Jarrodsz** [reply]
Ah right that was it i dont think i have xcode 15 yet. Will sure do its more to study the code your repro had inplemented some parts i was very curious about how they are done.

Its a bit steep for me as newbie so i tey some cool things basica like tray setting menu in a minimal new peoject for leari g. If i comeup with an awesome idea i make a pr but not sure what could be inproved. I like rectangle a lot perhaps thats a good inspiration?

i really love the configurable icon thingy thats just awesome engineering! Kind regards 🙏
